activerecord/lib/active_record/integration.rb

```diff
@@ -161,8 +161,8 @@ def to_param(method_name = nil)
         end
       end
 
-      def collection_cache_key(collection = all, timestamp_column = :updated_at) # :nodoc:
-        collection.send(:compute_cache_key, timestamp_column)
+      def collection_cache_key(collection = all, id_column: :id, timestamp_column: :updated_at) # :nodoc:
```

Comment:
Default for `id_column` is hardcoded to `:id`, should it be `primary_key`?

Reply:
Thanks for catching that. It should be consistent throughout, I've refactored accordingly.